Dockerfile

```diff
@@ -20,87 +20,146 @@
+# FROM ubi9/nodejs-18:1 AS builder
+#/ Downstream uncomment
+
+# hadolint ignore=DL3002
 USER 0
 
 # Install isolated-vm dependencies
-RUN dnf update -y && \
-  dnf install -y zlib-devel brotli-devel
+# hadolint ignore=DL3041
+RUN dnf install -y -q --allowerasing --nobest nodejs-devel nodejs-libs \
+  # already installed or installed as deps: 
+  openssl openssl-devel ca-certificates make cmake cpp gcc gcc-c++ zlib zlib-devel brotli brotli-devel python3 nodejs-packaging && \
+  dnf update -y && dnf clean all
 
 # Env vars
 ENV YARN=./.yarn/releases/yarn-1.22.19.cjs
 
 # Upstream sources
 # Downstream comment
 ENV EXTERNAL_SOURCE=.
+ENV EXTERNAL_SOURCE_NESTED=.
 ENV CONTAINER_SOURCE=/opt/app-root/src
 #/ Downstream comment
-
 # Downstream sources
 # Downstream uncomment
 # ENV EXTERNAL_SOURCE=$REMOTE_SOURCES/upstream1/app
+# ENV EXTERNAL_SOURCE_NESTED=$EXTERNAL_SOURCE/distgit/containers/rhdh-hub
+# # /remote-source/
 # ENV CONTAINER_SOURCE=$REMOTE_SOURCES_DIR
 #/ Downstream uncomment
 
 WORKDIR $CONTAINER_SOURCE/
-COPY $EXTERNAL_SOURCE/.yarn ./.yarn
-COPY $EXTERNAL_SOURCE/.yarnrc.yml ./
+COPY $EXTERNAL_SOURCE_NESTED/.yarn ./.yarn
+COPY $EXTERNAL_SOURCE_NESTED/.yarnrc.yml ./
 RUN chmod +x $YARN
 
-# Downstream uncomment
-# COPY $EXTERNAL_SOURCE/../cachito.env $EXTERNAL_SOURCE/../registry-ca.pem ./
-#/ Downstream uncomment
-
 # Stage 2 - Install dependencies
+# Downstream comment
 FROM skeleton AS deps
+#/ Downstream comment
+# Downstream comment
+# # debugging paths to files
+# COPY $REMOTE_SOURCES/ ./
+# hadolint ignore=SC2086
+# RUN for d in $(find / -name ".npmrc" -o -name ".yarnrc" 2>/dev/null); do echo; echo "==== $d ===="; cat $d; done
+# # ls -la ./ upstream1 upstream1/app/ upstream1/app/distgit/containers/ upstream1/app/distgit/containers/rhdh-hub/ || true
+# # debug contents of /remote-source/
+#  echo "###### /tmp/tls-ca-bundle.pem ######>>"; cat /tmp/tls-ca-bundle.pem; echo "<<###### /tmp/tls-ca-bundle.pem ######"
+#  echo "###### $CONTAINER_SOURCE/registry-ca.pem ######>>"; cat $CONTAINER_SOURCE/registry-ca.pem; echo "<<###### $CONTAINER_SOURCE/registry-ca.pem ######"
+#/ Downstream comment
 
-COPY $EXTERNAL_SOURCE/package.json $EXTERNAL_SOURCE/yarn.lock ./
-COPY $EXTERNAL_SOURCE/packages/app/package.json ./packages/app/package.json
-COPY $EXTERNAL_SOURCE/packages/backend/package.json ./packages/backend/package.json
+COPY $EXTERNAL_SOURCE_NESTED/package.json $EXTERNAL_SOURCE_NESTED/yarn.lock ./
+COPY $EXTERNAL_SOURCE_NESTED/packages/app/package.json ./packages/app/package.json
+COPY $EXTERNAL_SOURCE_NESTED/packages/backend/package.json ./packages/backend/package.json
 
 # see https://docs.engineering.redhat.com/pages/viewpage.action?pageId=228017926#UpstreamSources(Cachito,ContainerFirst)-CachitoIntegrationfornpm
 # Downstream uncomment
-# RUN source $CONTAINER_SOURCE/cachito.env && $YARN config set nodedir /usr; $YARN config set unsafe-perm true; # $YARN config list --verbose
+# COPY $REMOTE_SOURCES/upstream1/cachito.env \
```

Comment:
this whole block can be made smaller if upstream prefers the debugging stuff removed and streamlined.